src/movegen.cpp

```diff
@@ -43,6 +43,11 @@ namespace {
             b ^= to - pawn_push(us);
         if (pos.variant()->arrowGating)
             b &= moves_bb(us, type_of(pos.piece_on(from)), to, pos.pieces() ^ from);
+        if (pos.variant()->staticGating)
```

Comment:
Please note that move generation and pseudo-legal move validation always need to be in sync in order to ensure that moves coming from the TT are actually valid. Therefore statements analogous to https://github.com/ianfab/Fairy-Stockfish/blob/a0b277d7abfe6094bd44da4b113b00a94f0325a1/src/position.cpp#L1218-L1219 should be added to pseudo-legal validation.